### PR TITLE
Added version command

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Added version command. It should return flytectl and flyteadmin version. 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/173

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package version
+
+import (
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/admin"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// This module provides the ability to inject Build (git sha) and Version information at compile time.
```
Comment (kumare3) [inline version/version.go:10]: can we not re-use directly from flytestdlib - it is a dependency

**Comment by evalsocket** [comment]
@kumare3 pr is ready you can review it.

File: version/version.go (hunk 1 above)
Comment (evalsocket) [reply version/version.go:11]: This version package is for flytectl. And we are setting the value through ldflags

Comment (EngHabu) [reply version/version.go:11]: You should be able to use stdlib.. all it has is a copy of this file... you can set it there in LDFlags, then just read it from the same file when the version command is invoked... It'll not be setting the version of stdlib or anything... it's still just setting the version in this binary...

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,15 @@
 export REPOSITORY=flytectl
 include boilerplate/lyft/golang_test_targets/Makefile
 
+GIT_VERSION := $(shell git describe --always --tags)
+GIT_HASH := $(shell git rev-parse --short HEAD)
+TIMESTAMP := $(shell date '+%Y-%m-%d')
+PACKAGE ?=github.com/flyteorg/flyteadmin
```
Comment (EngHabu) [inline Makefile:7]: Why does this say flyteadmin?

**Comment by evalsocket** [comment]
@EngHabu currently we are printing output of version command in log.

```bash
➜  flytectl git:(version) ✗ ./bin/flytectl version
Warn: No metricsProvider set for the workqueue
INFO[0000] Using config file: [config.yaml]             
INFO[0000] Config section [adminutils] updated. No update handler registered.  src="viper.go:318"
INFO[0000] Config section [root] updated. No update handler registered.  src="viper.go:318"
INFO[0000] Config section [logger] updated. Firing updated event.  src="viper.go:320"
{"json":{"src":"viper.go:320"},"level":"info","msg":"Config section [admin] updated. Firing updated event.","ts":"2021-03-27T01:15:03+05:30"}
{"json":{"src":"viper.go:318"},"level":"info","msg":"Config section [storage] updated. No update handler registered.","ts":"2021-03-27T01:15:03+05:30"}
{"json":{"src":"client.go:30"},"level":"info","msg":"Initialized Admin client","ts":"2021-03-27T01:15:03+05:30"}
{"json":{},"level":"info","msg":"------------------------------------------------------------------------","ts":"2021-03-27T01:15:03+05:30"}
{"json":{},"level":"info","msg":"App [flytectl], Version [43f10a7], BuildSHA [43f10a7], BuildTS [2021-03-27 01:15:03.871811 +0530 IST m=+0.081155335]","ts":"2021-03-27T01:15:03+05:30"}
{"json":{},"level":"info","msg":"------------------------------------------------------------------------","ts":"2021-03-27T01:15:03+05:30"}
{"json":{},"level":"info","msg":"------------------------------------------------------------------------","ts":"2021-03-27T01:15:03+05:30"}
{"json":{},"level":"info","msg":"App [flyteadmin], Version [v0.3.7-47-gfa70a14], BuildSHA [fa70a14], BuildTS [2021-03-26 19:41:13.043553376 +0000 UTC m=+0.352678459]","ts":"2021-03-27T01:15:03+05:30"}
{"json":{},"level":"info","msg":"------------------------------------------------------------------------","ts":"2021-03-27T01:15:03+05:30"}
```

File: cmd/version/version.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			version.LogBuildInformation("flytectl")
+			PrintVersion("flyteadmin", v)
+		},
+	}
+
+	return versionCmd
+}
+
+func PrintVersion(appName string, res *admin.GetVersionResponse) {
+	logrus.Info("------------------------------------------------------------------------")
```
Comment (EngHabu) [inline cmd/version/version.go:55]: plz use `logger` from stdlib...

**Review by EngHabu — COMMENTED** [review]
You can also set formatter to Text if the goal is to see this in a terminal (will look nicer):
https://github.com/flyteorg/flytestdlib/blob/master/logger/config.go#L27

File: cmd/version/version.go (hunk 3 above)
Comment (evalsocket) [reply cmd/version/version.go:55]: @EngHabu I can make these changes....but it's a command....we can't add its output in the log. I found that flytestdlib version package only prints version in log(We need to set the log level to print output).

File: cmd/version/version.go [hunk 4]
```diff
@@ -2,13 +2,12 @@ package version
...
 	"context"
+	"encoding/json"
 	"fmt"
-	"os"
 
 	adminclient "github.com/flyteorg/flyteidl/clients/go/admin"
 	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
-	"github.com/flyteorg/flytestdlib/version"
-	"github.com/sirupsen/logrus"
+	adminversion "github.com/flyteorg/flytestdlib/version"
```
Comment (wild-endeavor) [inline cmd/version/version.go:10]: why is this called "adminversion"?  makes it sound like it's the version of Flyte Admin.

Comment (evalsocket) [reply cmd/version/version.go:10]: ok

**Review by EngHabu — APPROVED** [review]

**Pull request merged**